### ISSUE TITLE
Compatibility with YAML anchors

### ISSUE THREAD

**Issue by xaviml** [AppDaemon/appdaemon#1003]
## Problem

I am opening this issue for a **feature request proposal**. The PyYAML version that AppDaemon uses allows YAML anchors, which is great to reduce code duplication between application configurations. Let's say we have these 2 application configurations:

```yaml
example1:
  module: my_module
  class: my_class
  attr1: common_attr
  attr2: common_attr
  attr3: common_attr
  attr4:  different_value1

example2:
  module: my_module
  class: my_class
  attr1: common_attr
  attr2: common_attr
  attr3: common_attr
  attr4:  different_value2
```

Now with AppDaemon you can reduce this to the following:

```yaml
common_config: &common_config
  module: my_module
  class: my_class
  attr1: common_attr
  attr2: common_attr
  attr3: common_attr

example1:
  <<: *common_config
  attr4: different_value1

example2:
  <<: *common_config
  attr4: different_value2
```

This works perfectly now, however, PyYAML does not remove the `common_config` from the parsed YAML and AppDaemon tries to run the app `common_config`. What I have seen in other projects like [GitLab CICD](https://docs.gitlab.com/ee/ci/yaml/#hide-jobs), for example, is that adding a dot (.) to the beginning of the config ignores that config.

## Feature request

So the feature request would be to ignore configurations starting with a dot, so the following configuration would work as expected:

```yaml
.common_config: &common_config
  module: my_module
  class: my_class
  attr1: common_attr
  attr2: common_attr
  attr3: common_attr

example1:
  <<: *common_config
  attr4: different_value1

example2:
  <<: *common_config
  attr4: different_value2
```

## Addition information

What I am proposing is inserting another if condition in [here](https://github.com/AppDaemon/appdaemon/blob/cd0fa4cd83480480287b6970b9c47e07330cf8cd/appdaemon/app_management.py#L361) like:

```python
and not app.startswith(".")
```

Let me know what you guys think and thank you for this awesome project! :)

**Comment by Odianosen25** [comment]
@xaviml this is a very great suggestion in my opinion, can you just make a PR for it? It looks relatively straightforward as you said.

Regards

**Comment by xaviml** [comment]
This issue can be closed since it was released in AppDaemon v4.0.6.
